src/components/DayPickerRangeController.jsx

```diff
@@ -213,11 +213,12 @@ export default class DayPickerRangeController extends React.PureComponent {
 
     this.isTouchDevice = isTouchDevice();
     this.today = moment();
+    this.cachedProps = {};
```

Comment:
This will break use cases; it’s not guaranteed that these callbacks’ return values are memoizable.

Reply:
I think, these callbacks return values can be memoizable during render time at least, but it's up to you to decide.

Reply:
During a single render, that seems reasonable, but also I’m not sure what the value is of adding this optimization.

Reply:
Value is performance improvement. 3x less calls of isDayBlocked() and isOutsideRange(). If user has DayPickerRangeController with both callbacks used, then 90(3 months)*2(methods)*3(not cached) = 540 calls performed. After the improvement with return values cache, calls count decreases to 180.

Reply:
hmm. this seems like a bit more complexity than should be necessary for per-render caching. could we split that out into a separate PR, and keep this one just for the bug fix and regression tests?

Reply:
Moved cache commit to PR #2147